Frontend/library/src/Util/FileUtil.ts

```diff
@@ -58,27 +58,32 @@ export class FileUtil {
         // If we haven't received the initial setup instructions, return
         if (!file.receiving) return;
 
-        // Extract the total size of the file (across all chunks)
-        file.size = Math.ceil(
-            new DataView(view.slice(1, 5).buffer).getInt32(0, true) /
-                16379 /* The maximum number of payload bits per message*/
+        const typeSize = 1;
+        const intSize = 4;
+        const maxMessageSize = 16 * 1024;
+        const headerSize = typeSize + intSize;
+        const maxPayloadSize = maxMessageSize - headerSize;
+
+        // Calculate total number of chunks from the total file size
+        file.chunks = Math.ceil(
+            new DataView(view.slice(typeSize, headerSize).buffer).getInt32(0, true) / maxPayloadSize
         );
 
         // Get the file part of the payload
-        const fileBytes = view.slice(1 + 4);
+        const fileBytes = view.slice(headerSize);
 
         // Append to existing data that holds the file
         file.data.push(fileBytes);
 
         // Uncomment for debug
-        Logger.Info(`Received file chunk: ${file.data.length}/${file.size}`);
+        Logger.Info(`Received file chunk: ${file.data.length}/${file.chunks}`);
 
-        if (file.data.length === file.size) {
+        if (file.data.length === file.chunks) {
             file.receiving = false;
             file.valid = true;
             Logger.Info('Received complete file');
             const transferDuration = new Date().getTime() - file.timestampStart;
-            const transferBitrate = Math.round((file.size * 16 * 1024) / transferDuration);
+            const transferBitrate = Math.round((file.chunks * maxPayloadSize) / transferDuration);
```

Comment:
Think `maxPayloadSize` should be `maxMessageSize`. The transfer bitrate calculation should still include the 5 byte header because even though it wasn't used in the file contents, it was still transferred data.